exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillSqlWorker.java

```diff
@@ -183,4 +183,20 @@ private static PhysicalPlan getQueryPlan(QueryContext context, String sql, Point
 
     return handler.getPlan(sqlNode);
   }
+
+  //Wrap with Auto Limit
+  private static SqlNode wrapWithAutoLimit(SqlConverter parser, QueryContext context, String sql) {
+    SqlNode sqlNode = parser.parse(sql);
```

Comment:
Please replace this code with the following to avoid parsing the query several times.
```suggestion
    SqlNode sqlNode = parser.parse(sql);

    int fetch = context.getAutoLimitRowCount();

    if (sqlNode.getKind().belongsTo(SqlKind.QUERY)) {
      if (sqlNode.getKind() == SqlKind.ORDER_BY) {
        // merge SqlOrderBy
        SqlOrderBy orderBy = (SqlOrderBy) sqlNode;
        if (orderBy.fetch != null) {
          fetch = Math.min(Integer.valueOf(orderBy.fetch.toString()), fetch);
        }
        sqlNode = new SqlOrderBy(orderBy.getParserPosition(), orderBy.query, orderBy.orderList, orderBy.offset,
            SqlLiteral.createExactNumeric(Integer.toString(fetch), SqlParserPos.ZERO));
      } else {
        sqlNode = new SqlOrderBy(SqlParserPos.ZERO, sqlNode, SqlNodeList.EMPTY, null,
            SqlLiteral.createExactNumeric(Integer.toString(fetch), SqlParserPos.ZERO));
      }
    }
```

Reply:
This helps a lot! I've no knowledge of the Calcite SqlNodes, so this snippet helps.